### PR TITLE
Fix custom expressions with multiple arguments - contains, does-not-contain, starts-with, ends-with

### LINKED ISSUE #49304: Editing custom expressions for Contains filter with multiple values causes an error
### Describe the bug

If you use contains/endwith/startswith filter with multiple values, and then attempt to edit the custom expression, the filter block in the query builder breaks. It works fine when you only have one value in the contains/endwith/startswith filter 

https://github.com/user-attachments/assets/1211a637-3254-4383-ad6e-27707d037fa4

### To Reproduce

1. Start a query builder question from "Products"
2. Add a filter on `Category`: Contains `Gad`, `Wid`
Image

4. Click "Add filter". (Note: another bug it seems that you have to click "Add filter" twice)
5. Once the filter is added, click on it to edit
6. In the filter editing thing, click on the "Back" button to get back to the one that has "Custom expression"
7. Click on "Custom expression"
8. Error:
Image

### Expected behavior

Opens a custom expression editor with the custom expression for the filter filled in, like it does when there's one value in the filter

### Logs

_No response_

### Information about your Metabase installation

51.1 and master `c3f9a20`

### Severity

P3?

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ranquild**
Fixes https://github.com/metabase/metabase/issues/49304

How to verify:
- New -> Question -> Products -> Filter -> Custom expression. Try:
- `contains([Ean], "A", "B")`
- `contains([Ean], "A", "B", "C")`
- `contains([Ean], "A", "B", "case-insensitive")`
- `contains([Ean], "A", "B", "C", "case-insensitive")`
- These clauses should be picked by the widget; try changing them both via the widget and the expression editor and see if options are picked up correctly.

### CODE REVIEW

**Review by mngr — APPROVED** [review]
Looks good to me!

File: frontend/src/metabase-lib/filter.ts [hunk 1]
```diff
@@ -615,7 +615,7 @@ function getStringFilterOptions(
 ): StringFilterOptions {
   const operators: ReadonlyArray<string> = STRING_FILTER_OPERATORS_WITH_OPTIONS;
   const supportsOptions = operators.includes(operator);
-  return supportsOptions ? { "case-sensitive": false, ...options } : {};
+  return supportsOptions ? { "case-sensitive": true, ...options } : {};
```
Comment (ranquild) [inline frontend/src/metabase-lib/filter.ts:618]: When this is not set the QP defaults to `true` and not `false`!

File: frontend/src/metabase-lib/v1/expressions/recursive-parser.js [hunk 2]
```diff
@@ -319,6 +324,36 @@ export const adjustOffset = tree =>
...
+ ["contains", {}, ["field", 1, null], "A", "B"]
+ ["contains", {"case-sensitive": false}, ["field", 1, null], "A", "B"]
+
+ By default, the expression parser adds the options object as the last operand,
+ so we need to adjust its position here or insert an empty options object if
+ there is none.
+*/
+export const adjustMultiArgOptions = tree =>
+  modify(tree, node => {
+    if (Array.isArray(node)) {
```
Comment (uladzimirdev) [inline frontend/src/metabase-lib/v1/expressions/recursive-parser.js:342]: 🤞

File: e2e/test/scenarios/custom-column/custom-column-reproductions.cy.spec.js [hunk 3]
```diff
@@ -1179,3 +1180,101 @@ describe("issue 49882", () => {
...
+    openNotebook();
+
+    cy.log(
+      "add a filter using a filter widget and check that it is rendered in the expression editor",
+    );
+    getNotebookStep("data").button("Filter").click();
+    popover().findByText("Category").click();
+    selectFilterOperator("Contains");
+    popover().within(() => {
+      cy.findByPlaceholderText("Enter some text").type("gadget,widget");
```
Comment (uladzimirdev) [inline e2e/test/scenarios/custom-column/custom-column-reproductions.cy.spec.js:1207]: shouldn't we test parts of the filters and startsWith, endsWith?

at least steps from the issue show the problem with partial words

**Review by uladzimirdev — APPROVED** [review]

**Comment by uladzimirdev** [comment]
I tested locally, works fine

File: e2e/test/scenarios/custom-column/custom-column-reproductions.cy.spec.js (hunk 3 above)
Comment (ranquild) [reply e2e/test/scenarios/custom-column/custom-column-reproductions.cy.spec.js:1207]: I added unit tests for all other functions

**Pull request merged**